### PR TITLE
Update to Gradle 8.10

### PR DESCRIPTION
**Pull request by jamesfredley**
Requires the following PRs to be merged and snapshots to be published.  Then passes all tests.

https://github.com/grails/grails-core/pull/13622 - merged already
https://github.com/grails/grails-core/pull/13630 - duplicate messageSource
https://github.com/grails/grails-views/pull/581 - duplicate messageSource

 -x test -x integrationTest can be removed from gradle.yml once the above PRs are completed.

### CODE REVIEW

**Comment by jamesfredley** [comment]
Still has failing tests, but assembles and runs, which is progress.  This will allow a snapshot to be published.

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -22,4 +22,21 @@ dependencies {
...
+
+def groovyVersion = project.rootProject
+        .extensions
+        .getByType(VersionCatalogsExtension.class)
+        .named("libs")
+        .findVersion("groovy")
+        .get()
+        .displayName
+
+configurations.configureEach {
```
Comment (codeconsole) [inline buildSrc/build.gradle:35]: why is this needed?

File: examples/pubsub-demo/build.gradle [hunk 2]
```diff
@@ -48,4 +50,17 @@ dependencies {
 // Don't need groovydoc for example project
 tasks.named('groovydoc') {
     enabled = false
-}
\ No newline at end of file
+}
+
+bootJar {
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
```
Comment (codeconsole) [inline examples/pubsub-demo/build.gradle:56]: why?

Comment (jamesfredley) [reply examples/pubsub-demo/build.gradle:56]: To work around duplicates that exist during the build.  These were new after upgrade to gradle 8.10, but DuplicatesStrategy is set in other Grails projects.

File: buildSrc/build.gradle (hunk 1 above)
Comment (jamesfredley) [reply buildSrc/build.gradle:35]: A number of dependencies were still pulling in org.codehaus.groovy at the time.
